agent/tool-scripts/README.md

```diff
@@ -0,0 +1,219 @@
+`pbench-tool-meister-start` was invoked), the captured data is stored directly
+into the local `${benchmark_result_dir}` directory.  Tool Meisters running on
+hosts remote from the driving benchmark host direct the tools to store their
+data in a temporary directory tracked by the Tool Meister.
+
+Each remote Tool Meister is responsible for gathering up the captured data for
+each tool and sending it back to the Tool Data Sink on the host driving the
+benchmark.
+
+The classic example of a "transient" tool is "perf record", where that command
+captures its data in a local directory, and has no interface in the tool
+itself to send it remotely.
+
+### The `external-data-source` Tool
+
+The `external-data-source` tool provides a way for recording the URL to a data
+source external to the benchmark environment.  For example, the benchmark
+environment might be using a third-party data collection tool which has its
+own web interface to review data.  This tool gives the user a way to record
+the URL associated with the benchmark.
+
+This tool does not provide any `datalog` or `postprocess` scripts, and no
+command to collect data is invoked.
+
+## The `datalog` and `postprocess` Directories for "transient" Tools
+
+For each transient tool, there must be a `datalog` script which is responsible
+for invoking the command that generates the data for that tool.  These scripts
+are stored in the `datalog` sub-directory named `<tool>-datalog`.
+
+Each transient tool can optionally provide a "stop-postprocess" script which
+handles the required steps for properly capturing all of the relevant data and
+meta-data for that tool.  These scripts are stored in the "postprocess"
+sub-directory named `<tool>-stop-postprocess`.  The `stop-postprocess` script
+is executed on the host running the tool, after that tool is stopped
+successfully, but before reporting back to the caller the tool has stopped.
+
+Similarly, each transient tool can also optionally provide a "postprocess"
+script which handles post-processing steps for the captured data to apply any
+required or desired transformations.  The script is executed on the local host
+when the `pbench-postprocess-tools` script is invoked.
+
+## Installation Steps for the `node-exporter` Tool
+
+In cases where your operating system distribution does not provide a package
+for installing the Prometheus `node_exporter` command, consider following the
+steps below on the hosts from which you want to collect Prometheus
+`node_exporter`, and/or refer to the `node_exporter` [GitHub source
+repository](https://github.com/prometheus/node_exporter).
+
+ 1. `wget https://github.com/prometheus/node_exporter/releases/download/v1.0.1/node_exporter-1.0.1.linux-amd64.tar.gz`  (latest version, change version number as desired)
+ 1. `tar xvfz node_exporter-*.*-amd64.tar.gz`
+ 1. `mv node_exporter-1.0.1.linux-amd64 node_exporter`  (not necessary but nice)
+ 1. Test that it's working:
```

Comment:
All of them are "1.", followed by "4.1" and "4.2"

Reply:
Good catch ...